collect_app/src/main/java/org/odk/collect/android/utilities/FormDefCache.java

```diff
@@ -108,17 +109,11 @@ private static File getCacheFile(File formXml) {
                 FileUtils.getMd5Hash(formXml) + ".formdef");
     }
 
-    private static FormDef deserializeFormDef(File serializedFormDef) {
+    private static FormDef deserializeFormDef(File serializedFormDef) throws Exception {
```

Comment:
Do we need readCache() and deserializeFormDef() looks like the same code is duplicated in javarosa `org.javarosa.benchmarks.utils.FormDefCache`?

Reply:
Yes, this is client code. The package you point to is for benchmarks and that code was copied from Collect. You're right that the duplication is weird and I wonder whether we should rework the benchmark a bit on the JR side. I don't think there are changes to make here, though.

Reply:
Well, I don't know. Maybe some of it should move down to JR. Seems like a consideration for another time, though!